sklearn/cluster/hierarchical.py

```diff
@@ -751,6 +751,15 @@ def __init__(self, n_clusters=2, affinity="euclidean",
         self.affinity = affinity
         self.pooling_func = pooling_func
 
+    @property
+    def n_components_(self):
+        warnings.warn("The `n_components` parameter changed to"
```

Comment:
Please use the `@deprecated` decorator as suggested in the [dev guide](https://scikit-learn.org/stable/developers/contributing.html#deprecation)

Reply:
Hi Nicolas thanks for the feedback, apologies I had not seen that in the dev guide - commit #03a6682